polaris-core/src/main/java/org/apache/polaris/core/persistence/BasePersistence.java

```diff
@@ -0,0 +1,435 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.persistence;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.apache.polaris.core.PolarisCallContext;
+import org.apache.polaris.core.entity.PolarisBaseEntity;
+import org.apache.polaris.core.entity.PolarisChangeTrackingVersions;
+import org.apache.polaris.core.entity.PolarisEntityActiveRecord;
+import org.apache.polaris.core.entity.PolarisEntityCore;
+import org.apache.polaris.core.entity.PolarisEntityId;
+import org.apache.polaris.core.entity.PolarisEntityType;
+import org.apache.polaris.core.entity.PolarisGrantRecord;
+import org.apache.polaris.core.entity.PolarisPrincipalSecrets;
+import org.apache.polaris.core.storage.PolarisStorageConfigurationInfo;
+import org.apache.polaris.core.storage.PolarisStorageIntegration;
+
+/**
+ * Interface to the Polaris metadata store, allows to persist and retrieve all Polaris metadata like
+ * metadata for Polaris entities and metadata about grants between these entities which is the
+ * foundation of our role base access control model.
+ *
+ * <p>Note that APIs to the actual persistence store are very basic, often point read or write to
+ * the underlying data store. The goal is to make it really easy to back this using databases like
+ * Postgres or simpler KV store.
+ */
+public interface BasePersistence {
+  /**
+   * @param callCtx call context
+   * @return new unique entity identifier
+   */
+  long generateNewId(@Nonnull PolarisCallContext callCtx);
```

Comment:
Please specify ID uniqueness expectations.

Also, is ID generation permanent on return from this method (i.e. "committed" in RDBMS terms)?

Reply:
Done. Yes, unique within a realm and must be committed before returning it.